nmigen_boards/genesys2.py

```diff
@@ -0,0 +1,390 @@
+                 Subsignal("ba", Pins("AE9 AB10 AC11", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("ras_n", Pins("AE11", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("cas_n", Pins("AF11", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("we_n", Pins("AG13", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("dm", Pins("AD4 AF3 AH4 AF8", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("dq", Pins(
+                     "AD3 AC2 AC1 AC5 AC4 AD6 AE6 AC7 "
+                     "AF2 AE1 AF1 AE4 AE3 AE5 AF5 AF6 "
+                     "AJ4 AH6 AH5 AH2 AJ2 AJ1 AK1 AJ3 "
+                     "AF7 AG7 AJ6 AK6 AJ8 AK8 AK5 AK4", dir="io"),
+                     Attrs(IOSTANDARD="SSTL15_DCI", ODT="RTT_40")),
+                 Subsignal("dqs",
+                           DiffPairs(p="AD2 AG4 AG2 AH7",
+                                     n="AD1 AG3 AH1 AJ7", dir="io"),
+                           Attrs(IOSTANDARD="DIFF_SSTL15_DCI",
+                                 ODT="RTT_40")),
+                 Subsignal("clk",
+                           DiffPairs(p="AB9", n="AC9", dir="o"),
+                           Attrs(IOSTANDARD="DIFF_SSTL15_DCI")),
+                 Subsignal("cke", Pins("AJ9", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("odt", Pins("AK9", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15_DCI")),
+                 Subsignal("reset", PinsN("AG5", dir="o"),
+                           Attrs(IOSTANDARD="SSTL15")),
+                 Attrs(SLEW="FAST", OUTPUT_IMPEDANCE="RDRV_40_40")),
+        Resource("audio", 0,
+                 Subsignal("adc_sdata", Pins("AH19", dir="o")),
+                 Subsignal("dac_sdata", Pins("AJ19", dir="i")),
+                 Subsignal("bclk", Pins("AG18", dir="o")),
+                 Subsignal("lrclk", Pins("AJ18", dir="o")),
+                 Subsignal("mclk", Pins("AK19", dir="o")),
+                 Subsignal("scl", Pins("AE19", dir="io")),
+                 Subsignal("sda", Pins("AF18", dir="io")),
+                 Attrs(IOSTANDARD="LVCMOS18")),
+        Resource("oled", 0,
+                 Subsignal("data_cmd", PinsN("AC17", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS18")),
+                 Subsignal("reset", PinsN("AB17", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS18")),
+                 Subsignal("clk", Pins("AF17", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS18")),
+                 Subsignal("mosi", Pins("Y15", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS18")),
+                 Subsignal("vdd", PinsN("AG17", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS18")),
```

Comment:
Is it actually powered by the FPGA pin? If not it should be something like `vdd_en`.